sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -254,3 +255,46 @@ case class Pow(left: Expression, right: Expression)
       """
   }
 }
+
+case class Pmod(left: Expression, right: Expression) extends BinaryArithmetic {
+
+  override def symbol: String = "pmod"
+
+  override def toString: String = s"pmod($left, $right)"
+
+  protected def checkTypesInternal(t: DataType) =
+    TypeUtils.checkForNumericExpr(t, "pmod")
+
+  override lazy val resolved =
+    childrenResolved && checkInputDataTypes().isSuccess && !DecimalType.isFixed(dataType)
+
+  protected override def evalInternal(left: Any, right: Any) =
+    dataType match {
+      case IntegerType => pmod(left.asInstanceOf[Integer], right.asInstanceOf[Integer] )
+      case ShortType => pmod(left.asInstanceOf[Short], right.asInstanceOf[Short] )
+      case FloatType => pmod(left.asInstanceOf[Float], right.asInstanceOf[Float] )
+      case DoubleType => pmod(left.asInstanceOf[Double], right.asInstanceOf[Double] )
+      case _ => null
```

Comment:
according to [hive](https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFPosMod.java), I think we need to support `Decimal`...